core/src/main/java/io/undertow/server/RequestStatistics.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2022 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.undertow.server;
+
+public final class RequestStatistics {
+    private final HttpServerExchange exchange;
+
+    public RequestStatistics(HttpServerExchange exchange) {
+        this.exchange = exchange;
+    }
+
+    public long getBytesSent() {
+        return exchange.getResponseBytesSent();
+    }
+
+    public long getBytesReceived() {
+        return exchange.getRequestContentLength();
+    }
+
+    public long getStartTime() {
+        return exchange.getRequestStartTime();
+    }
+
+    public long getProcessingTime() {
+        long startTime = exchange.getRequestStartTime();
+        if (startTime > 0) {
+            return System.nanoTime() - startTime;
+        } else {
+            return -1;
+        }
+    }
```

Comment:
`exchange.getRequestStartTime()` comes from `System.nanoTime()` which may return negative values. I don't think `startTime > 0` is what we want. We could explicitly check for the `-1` sentinel value, though that is a valid result, or we could check whether `UndertowOptions.RECORD_REQUEST_START_TIME` is set.

Reply:
> `exchange.getRequestStartTime()` comes from `System.nanoTime()` which may return negative values. I don't think `startTime > 0` is what we want. We could explicitly check for the `-1` sentinel value, though that is a valid result, or we could check whether `UndertowOptions.RECORD_REQUEST_START_TIME` is set.

Ah.., you are right...

`-1` is the default value of `requestStartTime` in exchange when `UndertowOptions.RECORD_REQUEST_START_TIME` is not set. However, as you pointed out, there's a possibility that `System.nanoTime()` returns negative values. (And, this also means that `System.nanoTime()` can possibly return `-1`.., so using `-1` as the sentinel value is also not correct...).
So, it looks better to check whether `UndertowOptions.RECORD_REQUEST_START_TIME` is set.

Maybe, something like the following code is a better approach? (not yet compiled and tested by myself, though)

```
    public long getProcessingTime() {
        if (exchange.getConnection().getUndertowOptions().get(UndertowOptions.RECORD_REQUEST_START_TIME, false)) {
            return System.nanoTime() - exchange.getRequestStartTime();
        } else {
            return -1;
        }
    }
```

By the way, we already have a similar code in [io.undertow.server.ConnectorStatisticsImpl](https://github.com/undertow-io/undertow/blob/2.2.17.Final/core/src/main/java/io/undertow/server/ConnectorStatisticsImpl.java#L61-L62) that checks `startTime > 0`. And I noticed [io.undertow.attribute.ResponseTimeAttribute](https://github.com/undertow-io/undertow/blob/2.2.17.Final/core/src/main/java/io/undertow/attribute/ResponseTimeAttribute.java#L49-L50) just checks with the `-1` sentinel value. So, we may also need to update these codes to do the check processing consistently. Of course, I understand that updating these codes is not the scope of this pull request.

Reply:
Good catch. I've pushed an update. Is it a problem if System.nanoTime() returns a negative? The subtraction there will get us a negative return value. I guess the -1 in the else does that too, and this will just be a smaller number, so it's probably OK?